satellite.py

```diff
@@ -34,86 +34,140 @@
-constellations = [constellation
-                  for constellation, chosen in sampleset.first.sample.items()
-                  if chosen]
-
-tot = 0
-for constellation in constellations:
-    score = calculate_score(constellation, data)
-    print("Constellation: " + str(constellation) + ", Score: " + str(score))
-    tot += score
-print("Total Score: " + str(tot))
-print("Normalized Score (tot / # constellations): " + str((tot / data['num_constellations'])))
\ No newline at end of file
+def build_bqm(data, constellation_size):
+    """ Build the bqm for the problem."""
+
+    # don't consider constellations with average score less than score_threshold
+    score_threshold = .4
+
+    bqm = dimod.BinaryQuadraticModel.empty(dimod.BINARY)
+
+    # first we want to favor combinations with a high score
+    for constellation in itertools.combinations(range(data['num_satellites']), constellation_size):
+        # the score is the probability of at least one satellite in the constelation having line of sight over the target at any one time.
+        score = calculate_score(constellation, data)
+
+        # to make it smaller, throw out the combinations with a score below
+        # a set threshold
+        if score < score_threshold:
+            continue
+
+        # we subtract the score because we want to minimize the energy
+        bqm.add_variable(frozenset(constellation), -score)
+
+    # next we want to penalize pairs that share a satellite. We choose 2 because
+    # because we don't want it to be advantageous to pick both in the case that
+    # they both have 100% coverage
+    for c0, c1 in itertools.combinations(bqm.variables, 2):
+        if c0.isdisjoint(c1):
+            continue
+        bqm.add_interaction(c0, c1, 2)
+
+    # finally we wish to choose num_constellations variables. We pick strength of 1
+    # because we don't want it to be advantageous to violate the constraint by
+    # picking more variables
+    bqm.update(dimod.generators.combinations(bqm.variables, data['num_constellations'], strength=1))
+
+    return bqm
+
+def viz(constellations, data):
+    """ Visualize the solution"""
+    
+    angle = 2*math.pi / data["num_satellites"]
+
+    # display scatter plot data
+    plt.figure()
+    plt.title('Constellations')
+
+    s = 0
+    for c in constellations:
+        x = []
+        y = []
+        label = []
+        for satellite in c:
+            coverage = 1 - data["coverage"][str(satellite)]
+            label.append(satellite)
+            x.append(coverage*math.cos(s*angle))
+            y.append(coverage*math.sin(s*angle))
+            s += 1
+
+        x.append(x[0])
+        y.append(y[0])
+        label.append(label[0])
+        plt.plot(x, y, marker = 'o', markersize=1)
+
+    plt.scatter([0], [0], marker = '*', color='#7f7f7f', s=500)
+    plt.axis('off')
+    
+    # save plot
+    plt.savefig('constellations.png')
+
+    return
+
+if __name__ == '__main__':
+
+    args = read_in_args(sys.argv[1:])
```

Comment:
`sys.argv` is used automatically in `read_in_args`, not the value you pass here.